controller/search_spaces_blackbox_test.go

```diff
@@ -71,20 +72,81 @@ func (rest *TestSearchSpacesREST) TestSpacesSearchOK() {
-		{"with last page", args{offset(strconv.Itoa(len(idents) - 1)), limit(10), prefix + "TEST"}, expects{hasNoLinks("Next"), hasLinks("Prev")}},
-		{"with different values", args{offset("0"), limit(10), prefix + "TEST"}, expects{differentValues()}},
+		{"With uppercase fullname query", args{offset("0"), limit(10), prefix + "TEST_AB", nil}, expects{totalCount(1)}},
+		{"With lowercase fullname query", args{offset("0"), limit(10), prefix + "TEST_AB", nil}, expects{totalCount(1)}},
+		{"With uppercase description query", args{offset("0"), limit(10), "DESCRIPTION FOR " + prefix + "TEST_AB", nil}, expects{totalCount(1)}},
+		{"With lowercase description query", args{offset("0"), limit(10), "description for " + prefix + "test_ab", nil}, expects{totalCount(1)}},
+		{"with special chars", args{offset("0"), limit(10), "&:\n!#%?*", nil}, expects{totalCount(0)}},
+		{"with * to list all", args{offset("0"), limit(10), "*", nil}, expects{totalCountAtLeast(len(idents))}},
+		{"with multi page", args{offset("0"), limit(10), prefix + "TEST", nil}, expects{hasLinks("Next")}},
+		{"with last page", args{offset(strconv.Itoa(len(idents) - 1)), limit(10), prefix + "TEST", nil}, expects{hasNoLinks("Next"), hasLinks("Prev")}},
+		{"with different values", args{offset("0"), limit(10), prefix + "TEST", nil}, expects{differentValues()}},
+
+		{
+			name: "sorted by name ascending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("name"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_0", prefix+"TEST_1")},
+		},
+		{
+			name: "sorted by name descending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("-name"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_C", prefix+"TEST_B")},
+		},
+		{
+			name: "sorted by owner ascending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("owner"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_AB", prefix+"TEST_B")},
+		},
+		{
+			name: "sorted by owner descending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("-owner"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_AB", prefix+"TEST_B")},
+		},
+		{
+			name: "sorted by creation timestamp ascending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("created"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_A", prefix+"TEST_AB")},
+		},
+		{
+			name: "sorted by creation timestamp descending",
+			args: args{
+				pageOffset: offset("0"),
+				pageLimit:  limit(10),
+				q:          prefix + "TEST",
+				sort:       sortPtr("-created"),
+			},
+			expects: expects{sortedSpaces(prefix+"TEST_19", prefix+"TEST_18")},
```

Comment:
now it looks like the full possible set of tests.
Good work @surajssd :+1:

Reply:
I don't like these table-driven tests here. With a test fixture we would need the exact same lines but could create test data for each test and use just normal sub tests. That would be a whole lot more readable.